core/src/main/java/io/cucumber/core/filter/TagPredicate.java

```diff
@@ -2,31 +2,24 @@
 
 import io.cucumber.core.gherkin.Pickle;
 import io.cucumber.tagexpressions.Expression;
-import io.cucumber.tagexpressions.TagExpressionParser;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.function.Predicate;
 
-import static java.util.Collections.emptyList;
 import static java.util.Collections.singletonList;
 
 final class TagPredicate implements Predicate<Pickle> {
 
-    private final List<Expression> expressions = new ArrayList<>();
+    private final List<Expression> expressions;
 
-    TagPredicate(String tagExpression) {
-        this(tagExpression.isEmpty() ? emptyList() : singletonList(tagExpression));
+    TagPredicate(Expression tagExpression) {
```

Comment:
This constructor isn't used in production code. We might be able to get rid of it.

Reply:
Yeah, always a good idea to get rid of code that is not used anywhere.